### PR TITLE
Set explicit defaults w/ placeholders in the cdk.context.json

### PR DESCRIPTION
**Pull request by mikaylathompson**
### Description
[Describe what this change achieves]
* Category (Enhancement, New feature, Bug fix, Test fix, Refactoring, Maintenance, Documentation)
* Why these changes are required?
* What is the old behavior before changes and new behavior after changes?

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: deployment/cdk/opensearch-service-migration/cdk.context.json [hunk 1]
```diff
@@ -1,17 +1,17 @@
 {
   "demo-deploy": {
```
Comment (sumobrian) [inline deployment/cdk/opensearch-service-migration/cdk.context.json:2]: Replace demo-deploy name with <stage> or something else.

File: deployment/cdk/opensearch-service-migration/cdk.context.json [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
+      "auth": { "type": "none" }
+    },
+    "sourceCluster": {
+      "endpoint": "<SOURCE_CLUSTER_ENDPOINT>",
+      "auth": { "type": "none" }
+    },
+    "vpcId": "<VPC_ID>",
     "reindexFromSnapshotServiceEnabled": true,
+    "trafficReplayerServiceEnabled": true,
     "otelCollectorEnabled": true
```
Comment (sumobrian) [inline deployment/cdk/opensearch-service-migration/cdk.context.json:15]: I don't think this should be changed right now but this is an implementation detail. A user shouldn't have to know what OTEL is to use our solution.

**Review by sumobrian — CHANGES_REQUESTED** [review]

File: deployment/cdk/opensearch-service-migration/cdk.context.json (hunk 2 above)
Comment (mikaylathompson) [reply deployment/cdk/opensearch-service-migration/cdk.context.json:15]: Oh, yeah, I actually meant to pull this one out. It's set to true in the defaults, so they get it anyways

File: deployment/cdk/opensearch-service-migration/cdk.context.json [hunk 3]
```diff
@@ -1,17 +1,16 @@
...
-    "captureProxyESServiceEnabled": true,
+  "default": {
+    "stage": "<STAGE>",
+    "targetCluster": {
+      "endpoint": "<TARGET_CLUSTER_ENDPOINT>",
+      "auth": { "type": "none" }
+    },
+    "sourceCluster": {
+      "endpoint": "<SOURCE_CLUSTER_ENDPOINT>",
+      "auth": { "type": "none" }
```
Comment (AndreKurait) [inline deployment/cdk/opensearch-service-migration/cdk.context.json:10]: What about having options in here, even if not valid json? like

```
      "auth": { 
         "type": "none|basic|sigv4"
          // if basic
          "username"
          "password_from_secret_arn"
       }
    "vpcId": "<VPC_ID>",
```

Comment (mikaylathompson) [reply deployment/cdk/opensearch-service-migration/cdk.context.json:10]: I like this! Incorporating it now

**Review by sumobrian — APPROVED** [review]

**Pull request merged**